tests/test_supported_models.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+
+from templates.supported_models import MODEL_CONFIGS, ModelConfig, get_supported_models
+
+
+@pytest.fixture
+def sample_azure_deployment():
+    return {"type": "azure", "model_name": "gpt-35-turbo", "deployment_name": "azure"}
+
+
+@pytest.fixture
+def sample_openai_deployment():
+    return {"type": "openai", "model_name": "gpt-3.5-turbo", "deployment_name": ""}
+
+
+def test_model_config_structure():
+    assert isinstance(MODEL_CONFIGS, dict)
+
+    for model_name, config in MODEL_CONFIGS.items():
+        if isinstance(config, dict):
+            # Ensure both 'openai' and 'azure' keys exist in the dictionary
+            assert "openai" in config, f"{model_name} config should have an 'openai' entry."
+            assert "azure" in config, f"{model_name} config should have an 'azure' entry."
+            assert isinstance(
+                config["openai"], ModelConfig
+            ), f"The 'openai' entry for {model_name} should be a ModelConfig."
+            assert isinstance(
+                config["azure"], ModelConfig
+            ), f"The 'azure' entry for {model_name} should be a ModelConfig."
+
```

Comment:
Why is this restrictive? We can have Azure-hosted models that can't be found on openai.com.
In fact, our code is pretty flexible in that regard. It won't break if a different deployment type is given, it just won't find the specified model. Here we have the chance to specify exactly how strict we wanna be. Perhaps "azure" should always be part of that dict. And perhaps it should always point to a ModelConfig object. But why can't the openai bit be skipped? What if we decide to have a 3rd alternative there? Anyway, I don't really have a strong opinion, you can handle this however you see fit!

Reply:
You might be right, as we have enabled dynamic retrieval of Openai models we can have any kind of deployment we want for them, so perhaps it would make sense to actually remove both of them, as one might decide not to use azure hosted models at all in the future, but they want to handle same model-different identifiers issue we had using the same data type (e.g. llama models via Nvidia NIM API and HF inference).